### ISSUE TITLE
Switch to suppress verbose build output?

### ISSUE THREAD

**Issue by alanhogan** [storybooks/storybook#1872]
Our project added storybook's `start-storybook` command to our `Profile` so we can start it and a couple other servers with `nf` (Foreman).

This works great but builds cause an extreme amount of output in this manner:

<img width="722" alt="screen shot 2017-09-20 at 1 18 51 pm" src="https://user-images.githubusercontent.com/6808/30665768-041f1ab6-9e07-11e7-8933-ed5c7fc43a92.png">

Would it be possible for the CLI to take a switch (maybe `--quiet`) that would suppress verbose/progress update messages?

**Comment by ndelangen** [comment]
Hey Alan! We've had this feature request before, I think we should add it, would you like to help us?

It would involve a small CLI change and a webpack config change.
Should be fairly easy to implement, do you want to give it a show?

If so, please use the release/3.3 branch as your base 👍 🙇

**Comment by alanhogan** [comment]
I could take a look. Where in the codebase does `storybook-start` live / examine the flags it gets?

**Comment by ndelangen** [comment]
Here's our guide to contribute:
https://github.com/storybooks/storybook/blob/master/CONTRIBUTING.md

Here's the code points your need:

https://github.com/storybooks/storybook/tree/master/app/react-native/src/bin

https://github.com/storybooks/storybook/blob/master/app/react/src/server/index.js
https://github.com/storybooks/storybook/blob/master/app/react/src/server/build.js

https://github.com/storybooks/storybook/blob/master/app/vue/src/server/index.js
https://github.com/storybooks/storybook/blob/master/app/vue/src/server/build.js

`/app/.../src/server/config/webpack.config.js` should contain the webpack config that has the progress plugin, that you want to disable.

I hope this helps,

**Comment by Hypnosphi** [comment]
@ndelangen I changed all the links to point to `release/3.3` branch

**Comment by mmmeff** [comment]
How could we go about overriding the build output via webpack.config.js? I'm in full control mode and attempting to override it with `config.devServer = { stats: 'errors-only' };` but no dice.

**Comment by ndelangen** [comment]
@mmmeff What you need to do is filter out the ProgressPlugin if you want to remove it that way.

**Comment by mmmeff** [comment]
@ndelangen Since webpack's plugins field is defined as an array, removing that plugin is less than ideal. Here's my working but sloppy fix for disabling the progress output.

```
  config.plugins = config.plugins.filter((plugin) => {
    // Find and omit the plugin that looks like progress plugin.
    // This kinda sucks, but it works
    return !(
      Object.keys(plugin).length === 2
      && Object.keys(plugin).indexOf("profile") > -1
      && Object.keys(plugin).indexOf("handler") > -1
    )
  })
```

Got a better suggestion?

**Comment by Hypnosphi** [comment]
@mmmeff does `plugin.constructor.name !== 'ProgressPlugin' work?

**Comment by mmmeff** [comment]
@Hypnosphi That did the trick, thanks boss. For posterity, here's the configuration (simplified) to remove the progress plugin and quiet the log spam:

```
const genDefaultConfig = require('@storybook/react/dist/server/config/defaults/webpack.config.js')

module.exports = (baseConfig, env) => {
  const config = genDefaultConfig(baseConfig, env)
  config.plugins = config.plugins.filter(({ constructor }) => constructor.name !== "ProgressPlugin")

  return config
}

```

**Comment by alanhogan** [comment]
@mmmeff Is there a PR for that change? :D

**Comment by mmmeff** [comment]
@alanhogan No. You can add the above configuration to your own webpack.config.js in your .storybook directory though.

**Comment by alanhogan** [comment]
Oh you don’t say… cool… i'll have to give that a go, thanks!

**Comment by mmmeff** [comment]
@alanhogan These docs should help you out https://storybook.js.org/configurations/custom-webpack-config/#full-control-mode-default

**Comment by ndelangen** [comment]
Thanks for helping out @mmmeff !

**Comment by ndelangen** [comment]
Related: #1108

**Comment by Hypnosphi** [comment]
@storybooks/team how do you feel about making #3487 behavior the default one in 4.0, and introduce `--verbose` opt-it?

**Comment by Hypnosphi** [comment]
Released as [`4.0.0-alpha.4`](https://github.com/storybooks/storybook/releases/tag/v4.0.0-alpha.4)
